compiler/src/main/scala/dx/dxni/wdl/WdlNativeInterfaceGenerator.scala

```diff
@@ -77,7 +77,19 @@ case class WdlNativeInterfaceGenerator(wdlVersion: WdlVersion,
   private def wdlTypeFromDxClass(appletName: String,
                                  argName: String,
                                  ioClass: DxIOClass.Value,
-                                 isOptional: Boolean): WdlTypes.T = {
+                                 isOptional: Boolean): Option[WdlTypes.T] = {
+    if (ioClass == DxIOClass.Other) {
+      if (isOptional) {
+        logger.warning(s"ignoring applet ${appletName} optional non-file object input ${argName}")
+        return None
+      } else {
+        throw new Exception(
+            s"""|Cannot call applet ${appletName} from WDL, argument ${argName}
+                |has non-optional non-file object input ${argName}""".stripMargin
```

Comment:
Perhaps easier to read would be: "has required non-file object"

Reply:
This can be understood as primitive required types are also not allowed. Since we don't recognize the type, can we at least say: "argument ${name} of this type.."?

Reply:
the type is "other" which is a catch-all for non-file object types. I put it there but I don't think it's very helpful.